src/orquestra/vqa/algorithms/qcbm.py

```diff
@@ -0,0 +1,196 @@
+from functools import partial
+from numbers import Number
+from typing import Callable, Optional, cast
+
+import numpy as np
+from orquestra.opt.api import CostFunction, Optimizer
+from orquestra.opt.optimizers import ScipyOptimizer
+from orquestra.quantum.api.circuit_runner import CircuitRunner
+from orquestra.quantum.api.estimation import EstimateExpectationValues
+from orquestra.quantum.circuits import Circuit
+from orquestra.quantum.distributions import compute_clipped_negative_log_likelihood
+from orquestra.quantum.distributions.BAS_dataset import (
+    get_bars_and_stripes_target_distribution,
+)
+from orquestra.quantum.estimation import (
+    calculate_exact_expectation_values,
+    estimate_expectation_values_by_averaging,
+)
+from scipy.optimize import OptimizeResult
+
+from orquestra.vqa.ansatz.qcbm import QCBMAnsatz
+from orquestra.vqa.cost_function.qcbm_cost_function import create_QCBM_cost_function
+
+
+class QCBM:
+    def __init__(
+        self,
+        n_qubits: int,
+        n_layers: int,
+        optimizer: Optimizer,
+        estimation_method: EstimateExpectationValues,
+        n_shots: Optional[int] = None,
+    ) -> None:
+        """Class providing an easy interface to work with QCBM.
+
+        For new users, usage of "default" method is recommended.
+
+        Args:
+            n_layers: Number of layers for the ansatz.
+            n_qubits: Number of qubits for the circuit.
+            optimizer: Optimizer used to find optimal parameters
+            estimation_method: Method used for calculating expectation values of
+                the Hamiltonian.
+            n_shots: number of shots to be used for evaluation of expectation values.
+                For simulation with exact expectation value it should be None.
+        """
+        self.ansatz = QCBMAnsatz(n_layers, n_qubits, "all")
+        self._n_layers = n_layers
+        self._n_qubits = n_qubits
+        self.optimizer = optimizer
+        self.estimation_method = estimation_method
+        self._n_shots = n_shots
+
+    @classmethod
+    def default(
+        cls,
+        n_qubits: int,
```

Comment:
I think `n_qubits` would be inferred from the `target_distribution` 🤔 both here and in `__init__` .

Reply:
target_distribution is a dictionary right? so you mean like extracting it from the key value?

Reply:
Yes, that's what I mean :)